lang/cs/Org.Apache.REEF.IMRU/OnREEF/Driver/IMRUDriver.cs

```diff
@@ -286,6 +291,56 @@ public void OnNext(ICompletedTask completedTask)
             }
         }
 
+        public void OnNext(IFailedEvaluator value)
+        {
+            Logger.Log(Level.Info, "An evaluator failed, checking if it failed before context and service was submitted");
+            int currFailedEvaluators = Interlocked.Increment(ref _currentFailedEvaluators);
+
+            if (value.FailedTask != null && value.FailedTask.IsPresent())
+            {
+                Logger.Log(Level.Info, "An evaluator failed during task, cannot continue IMRU task");     
```

Comment:
This should be in a higher log level. Also, shouldn't this invoke some sort of crash logic?